cmd/promtool/backfill_test.go

```diff
@@ -0,0 +1,265 @@
+	require.Equal(t, 0, len(ss.Warnings()))
+
+	return result
+}
+
+func testBlocks(t *testing.T, blockpath string, mint, maxt int64, expectedSeries map[string][]tsdbutil.Sample) {
+	b, err := tsdb.OpenBlock(nil, blockpath, nil)
+	require.NoError(t, err)
+	q, err := tsdb.NewBlockQuerier(b, math.MinInt64, math.MaxInt64)
+	require.NoError(t, err)
+	series := queryblock(t, q, labels.MustNewMatcher(labels.MatchRegexp, "", ".*"))
+	require.Equal(t, expectedSeries, series)
+}
+func TestBackfill(t *testing.T) {
+	tests := []struct {
+		ToParse  string
+		IsOk     bool
+		Expected struct {
+			MinTime int64
+			MaxTime int64
+			Series  map[string][]tsdbutil.Sample
+		}
+	}{
+		{
+			ToParse: `# EOF`,
+			IsOk:    true,
+		},
+		{
+			ToParse: `# HELP http_requests_total The total number of HTTP requests.
+# TYPE http_requests_total counter
+http_requests_total{code="200"} 1021 1565133713989
+http_requests_total{code="400"} 1 1565133713990
+# EOF
+`,
+			IsOk: true,
+			Expected: struct {
+				MinTime int64
+				MaxTime int64
+				Series  map[string][]tsdbutil.Sample
+			}{
+				MinTime: 1565133713989000,
+				MaxTime: 1565133713990000,
+				Series:  map[string][]tsdbutil.Sample{"{__name__=\"http_requests_total\", code=\"200\"}": []tsdbutil.Sample{backfillSample{t: 1565133713989000, v: 1021}}, "{__name__=\"http_requests_total\", code=\"400\"}": []tsdbutil.Sample{backfillSample{t: 1565133713990000, v: 1}}},
+			},
+		},
+		{
+			ToParse: `# HELP http_requests_total The total number of HTTP requests.
+# TYPE http_requests_total counter
+http_requests_total{code="200"} 1022 1565133713989
+http_requests_total{code="400"} 2 1565133713990
+# EOF
+`,
+			IsOk: true,
+			Expected: struct {
+				MinTime int64
+				MaxTime int64
+				Series  map[string][]tsdbutil.Sample
+			}{
+				MinTime: 1565133713989000,
+				MaxTime: 1565133713990000,
+				Series:  map[string][]tsdbutil.Sample{"{__name__=\"http_requests_total\", code=\"200\"}": []tsdbutil.Sample{backfillSample{t: 1565133713989000, v: 1022}}, "{__name__=\"http_requests_total\", code=\"400\"}": []tsdbutil.Sample{backfillSample{t: 1565133713990000, v: 2}}},
+			},
+		},
+		{
+			ToParse: `# HELP http_requests_total The total number of HTTP requests.
+# TYPE http_requests_total counter
+http_requests_total{code="200"} 1023 1395066363000
+http_requests_total{code="400"} 3 1395066363000
+# EOF
+`,
+			IsOk: true,
+			Expected: struct {
+				MinTime int64
+				MaxTime int64
+				Series  map[string][]tsdbutil.Sample
+			}{
+				MinTime: 1395066363000000,
+				MaxTime: 1395066363000000,
+				Series:  map[string][]tsdbutil.Sample{"{__name__=\"http_requests_total\", code=\"200\"}": []tsdbutil.Sample{backfillSample{t: 1395066363000000, v: 1023}}, "{__name__=\"http_requests_total\", code=\"400\"}": []tsdbutil.Sample{backfillSample{t: 1395066363000000, v: 3}}},
+			},
+		},
+		{
+			ToParse: `# HELP something_weird Something weird
+		# TYPE something_weird gauge
+		something_weird{problem="infinite timestamp"} +Inf -3982045
```

Comment:
The value is infinite, not the timestamp. Which should be fine. Why is this test failing? I think it should be fine. What am I missing?

Reply:
Nothing, I thought some wrong logic and wrote this. Will remove